### ISSUE TITLE
ClickHouse Linting flags

### ISSUE THREAD

**Issue by lplplplp-lp** [sqlfluff/sqlfluff#7490]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Unexpected linting errors are being flagged in ClickHouse. e.g. no alias, this should be ok as there is only one model, indentation errors when indented correctly 

### Expected Behaviour

pass successfully

### Observed Behaviour

<img width="796" height="398" alt="Image" src="https://github.com/user-attachments/assets/015b97ec-47a5-4b70-b977-209850a3af4b" />

we are also seeing this in other models with other errors

<img width="1257" height="415" alt="Image" src="https://github.com/user-attachments/assets/0913f6fe-c922-4985-9052-7a41eb9e4743" />

<img width="806" height="438" alt="Image" src="https://github.com/user-attachments/assets/6f8aa74a-3127-4b39-a27a-6afd72d638bc" />

### How to reproduce

paste in the script and run `sqlfluff lint`
```
SELECT
  date_day AS date_key
  , DATETRUNC('week', date_day) AS week_key
  , DATETRUNC('month', date_day) AS month_key
  , DATEADD(DAY, -1, DATEADD(MONTH, 1, month_key)) AS month_end
  , DATETRUNC('quarter', date_day) AS quarter_key
  , DATEADD(DAY, -1, DATEADD(QUARTER, 1, quarter_key)) AS quarter_end
  , DATETRUNC('year', date_day) AS year_key
  , DAYOFWEEK(date_day) AS day_of_week
  , DAYOFMONTH(date_day) AS day_of_month
  , DAY(date_day) AS day_number
  , WEEK(date_day) AS week_number
  , MONTH(date_day) AS month_number
  , MONTHNAME(date_day) AS month_name
  , QUARTER(date_day) AS quarter_number
  , YEAR(date_day) AS year_number
FROM {{ ref('acq__datespine') }}
```

model 2
```
{{ 
  config(
    materialized='incremental'
    , unique_key='token_usage_id'
    , incremental_strategy='merge'
    , partition_by=['date_key']
    )
}}

WITH src AS (
  SELECT
    *
  FROM {{ source('runtime_logs', 'llm_token_usage') }}
  WHERE timestamp >= '{{ incremental_window() }}'
  QUALIFY ROW_NUMBER() OVER (PARTITION BY id ORDER BY timestamp DESC) = 1
)

SELECT
  id AS token_usage_id
  , timestamp AS token_usage_timestamp
  , CAST(timestamp AS DATE) AS date_key
  , client_id
  , event_verb
  , resource_name
  , resource_id
  , clickhouse_updated
  , service
  , provider
  , model
  , COALESCE(prompt_tokens, 0) AS prompt_tokens
  , COALESCE(completion_tokens, 0) AS completion_tokens
  , COALESCE(cached_tokens, 0) AS cached_tokens
  , conversation_id
  , module_id
  , rr_experiment
  , service_tier
  , prompt_id
  , current_timestamp() AS updated_at
FROM src

```

### Dialect

clickhouse

### Version

4.0.4

### Configuration

```
[sqlfluff]
dialect = clickhouse
templater = dbt


# The standard max_line_length is 80 in line with the convention of
# other tools and several style guides. Many projects however prefer
# something a little longer.
max_line_length = 160
# Ignore inline overrides (e.g. to test if still required)
disable_noqa = True

# Allow fix to run on files, even if they contain parsing errors
# Note altering this is NOT RECOMMENDED as can corrupt SQL
fix_even_unparsable = False
# Very large files can make the parser effectively hang.
# The more efficient check is the _byte_ limit check which
# is enabled by default. The previous _character_ limit check
# is still present for backward compatibility. This will be
# removed in a future version.
# Set either to 0 to disable.
large_file_skip_char_limit = 0
large_file_skip_byte_limit = 200000

##################################################
##################################################
# 1. Indentation
##################################################
##################################################
[sqlfluff:indentation]
tab_space_size = 2
indented_ctes = True
indented_joins = False
indented_on_contents = True
indented_then = False
indented_then_contents = True
indented_using_on = True
implicit_indents = allow
# If comments are found at the end of long lines, we default to moving
# them to the line _before_ their current location as the convention is
# that a comment precedes the line it describes.
trailing_comments = before

##################################################
##################################################
# 2. Layout
##################################################
##################################################
[sqlfluff:layout:type:comma]
spacing_before = touch
line_position = leading

[sqlfluff:layout:type:select_clause]
line_position = leading
spacing_after = touch

[sqlfluff:layout:type:where_clause]
line_position = leading
spacing_after = touch

[sqlfluff:layout:type:from_clause]
line_position = leading
spacing_after = touch

[sqlfluff:layout:type:join_clause]
line_position = leading
spacing_after = touch

[sqlfluff:layout:type:groupby_clause]
line_position = leading
spacing_after = touch

[sqlfluff:layout:type:orderby_clause]
# NOTE: Order by clauses appear in many places other than in a select
# clause. To avoid unexpected behaviour we use `leading` in this
# case rather than `alone`.
line_position = leading
spacing_after = touch

[sqlfluff:layout:type:having_clause]
line_position = leading
spacing_after = touch

[sqlfluff:layout:type:limit_clause]
line_position = leading
spacing_after = touch


##################################################
##################################################
# 3 Templater
##################################################
##################################################
[sqlfluff:templater]
unwrap_wrapped_queries = True

[sqlfluff:templater:jinja]
apply_dbt_builtins = True
library_path = dbt_packages

##################################################
##################################################
# 4. Rules
##################################################
##################################################

##################################################
# 4.1 capitalisation
##################################################
[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = upper
# Comma separated list of words to ignore for this rule
ignore_words = None
ignore_words_regex = None
[sqlfluff:rules:capitalisation.identifiers]
extended_capitalisation_policy = lower
# Comma separated list of words to ignore for this rule
ignore_words = None
ignore_words_regex = None
[sqlfluff:rules:capitalisation.functions]
extended_capitalisation_policy = upper
# Comma separated list of words to ignore for this rule
ignore_words = None
ignore_words_regex = None
[sqlfluff:rules:capitalisation.literals]
# Null & Boolean Literals
capitalisation_policy = upper
[sqlfluff:rules:capitalisation.types]
# Null & Boolean Literals
extended_capitalisation_policy = upper

##################################################
# 4.2 Conventions
##################################################
[sqlfluff:rules:convention.not_equal]
force_enable = True
[sqlfluff:rules:convention.coalesce]
# changes the use of 'NVL' or 'IFNULL' to 'COALESCE'
force_enable = True
[sqlfluff:rules:convention.select_trailing_comma]
# changes the use of 'NVL' or 'IFNULL' to 'COALESCE'
force_enable = True
select_clause_trailing_comma = forbid
[sqlfluff:rules:convention.count_rows]
prefer_count_0 = false
prefer_count_1 = false
[sqlfluff:rules:convention.is_null]
# when comparing against NULL, requires the IS or IS NOT comparison
force_enable = True
[sqlfluff:rules:convention.terminator]
#enforces semi-colon at the end of the file
multiline_newline = True
require_final_semicolon = False
[sqlfluff:rules:convention.statement_brackets]
#Top level statements should not be encased within brackets
force_enable = True
[sqlfluff:rules:convention.left_join]
#to keep consistent joins, we should not be using RIGHT JOIN - rewrites the query without
force_enable = True
[sqlfluff:rules:convention.blocked_words]
# Comma separated list of blocked words that should not be used
blocked_words = None
blocked_regex = None
match_source = False
[sqlfluff:rules:convention.quoted_literals]
# enforces single quotes for quotations
force_enable = True
preferred_quoted_literal_style = single_quotes
[sqlfluff:rules:convention.casting_style]
# Enforce standard CAST() style for type conversion (ClickHouse-friendly).
preferred_type_casting_style = cast



##################################################
# 4.3 Layouts
##################################################
[sqlfluff:rules:layout.spacing]
# enforces rule for spacing, checking for excessive/trailing 
# whitespace at the end of lines
force_enable = True
[sqlfluff:rules:layout.indent]
# enforces our rules for indentation
force_enable = True
[sqlfluff:rules:layout.operators]
# enforces operations that roll over lines to have the operator before the new line
force_enable = True
[sqlfluff:rules:layout.long_lines]
# enforces line lengths
force_enable = True
ignore_comment_clauses = False
ignore_comment_lines = False
[sqlfluff:rules:layout.cte_newline]
# enforces closing bracket for CTE to be on new line
force_enable = True
[sqlfluff:rules:layout.select_targets]
# enforces select targets to be on a new line
force_enable = True
wildcard_policy = multiple
[sqlfluff:rules:layout.select_modifiers]
# enforces select modifiers to be on same line as the SELECT clause i.e. SELECT DISTINCT
force_enable = True
[sqlfluff:rules:layout.set_operators]
# enforces SET operators such as UNION to be surrounded by a new line
force_enable = True
[sqlfluff:rules:layout.end_of_file]
# enforces a new line at the end of a file
force_enable = True
[sqlfluff:rules:layout.start_of_file]
# enforces that there can't be a new line or white space at the start of a file
force_enable = True


##################################################
# 4.4 References
##################################################
[sqlfluff:rules:references.from]
# enforces use of references only available in FROM/JOIN clauses
force_enable = True
[sqlfluff:rules:references.qualification]
# enforces declaration of reference when join is present
force_enable = True
[sqlfluff:rules:references.consistent]
# enforces no reference to a table if only a single table FROM clause
force_enable = True
single_table_references = qualified
[sqlfluff:rules:references.keywords]
# enforces keywords not to be used as references i.e. 'FROM table AS sum' is forbidden
quoted_identifiers_policy = none
unquoted_identifiers_policy = none
[sqlfluff:rules:references.special_chars]
# enforces what characters can be used within column names
# no special characters
allow_space_in_identifier = False
quoted_identifiers_policy = none
unquoted_identifiers_policy = none
[sqlfluff:rules:references.quoting]
# prefer identifiers and keywords to be UNQUOTED
prefer_quoted_identifiers = False
prefer_quoted_keywords = False

##################################################
# 4.5 Structure
##################################################
[sqlfluff:rules:structure.simple_case]
# Enforces simplicity of CASE statements
force_enable = True
[sqlfluff:rules:structure.unused_cte]
# Enforces usage of ALL CTE components, saves on processing and unnecessary code
force_enable = True
[sqlfluff:rules:structure.nested_case]
# Enforces simplicity of nesting CASE statements
force_enable = True
[sqlfluff:rules:structure.subquery]
# Enforces the rule that we don't use subqueries within FROM or JOIN conditions
force_enable = True
forbid_subquery_in = both
[sqlfluff:rules:structure.using]
# Enforces using ON rather than USING
force_enable = True
[sqlfluff:rules:structure.column_order]
# Enforces wildcard -> simple targets -> calculations and aggregates
force_enable = True
[sqlfluff:rules:structure.distinct]
# Enforces that DISTINCT must be applied to the whole table not just a column
force_enable = True

##################################################
# 4.5 Aliasing
##################################################
[sqlfluff:rules:aliasing.table]
# Ensures tables are aliased with 'AS'
aliasing = explicit
[sqlfluff:rules:aliasing.column]
# Ensures columns are aliased with 'AS'
aliasing = explicit
[sqlfluff:rules:aliasing.expression]
# Ensures expressions are aliased within the query 
allow_scalar = False
[sqlfluff:rules:aliasing.unique.table]
# Ensures table alias' are unique
force_enable = True
[sqlfluff:rules:aliasing.unused]
# Tables should not be aliased if that alias is not used
force_enable = True
[sqlfluff:rules:aliasing.length]
# Alias for tables should be no longer than 24 characters - simplicity
max_alias_length = 24



##################################################
# 4.6 Ambiguous
##################################################
[sqlfluff:rules:ambiguous.order_by]
# Ensures that Order by conditions are explicitly referenced
force_enable = True
[sqlfluff:rules:ambiguous.column_count]
# Ensures we don't use a wildcard to query data, it can be expensive as 
# as tables grow. Also likely to cause downstream errors in joins etc.
force_enable = True
[sqlfluff:rules:ambiguous.join]
# Ensures explicit reference of joins (currently INNER only)
force_enable = True
[sqlfluff:rules:ambiguous.column_references]
# Ensures implicit reference of columns in group/order by
group_by_and_order_by_style = implicit
[sqlfluff:rules:ambiguous.set_columns]
# Ensures set expressions return the same number of columns
force_enable = True


```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by peterbud** [comment]
Hi @lplplplp-lp thanks for opening this issue.

The linting warnings are the consequence of your config and it expected to work that way.

You have a config:
```ini
[sqlfluff:rules:references.consistent]
# enforces no reference to a table if only a single table FROM clause
force_enable = True
single_table_references = qualified
```

that forces to have fully qualified column names in your code. See [documentation](https://docs.sqlfluff.com/en/stable/reference/rules.html#column-references-should-be-qualified-consistently-in-single-table-statements)

Either you change your SQL to use fully qualified column references, like:

```sql
SELECT
    table1.date_day AS date_key,
    DATETRUNC('week', table1.date_day) AS week_key
FROM {{ ref('acq__datespine') }} AS table1
```

which WILL pass with your current config, or you change your `single_table_references` setting to `unqualified` or `consistent`, which will work with your current SQL.
